docs/examples/implementation-milestones-2.json

```diff
@@ -0,0 +1,211 @@
+                  "uri": "http://cpv.data.ac.uk/code-45233162.html"
+                }
+              ],
+              "quantity": 8,
+              "unit": {
+                "name": "Miles",
+                "value": {
+                  "amount": 137000,
+                  "currency": "GBP"
+                }
+              }
+            }
+          ],
+          "documents": [
+            {
+              "id": "0007",
+              "documentType": "notice",
+              "title": "Award notice",
+              "description": "Award of contract to build new cycle lanes in the centre of town to AnyCorp Ltd.",
+              "url": "http://example.com/tender-notices/ocds-213czf-000-00001-04.html",
+              "datePublished": "2010-05-10T10:30:00Z",
+              "format": "text/html",
+              "language": "en"
+            }
+          ]
+        }
+      ],
+      "contracts": [
+        {
+          "id": "ocds-213czf-000-00001-contract-01",
+          "awardID": "ocds-213czf-000-00001-award-01",
+          "title": "Contract to build new cycle lanes in the centre of town.",
+          "description": "Contract monitoring for cycle lane construction.",
+          "status": "active",
+          "period": {
+            "startDate": "2018-03-18T00:00:00Z",
+            "endDate": "2018-12-31T23:59:00Z"
+          },
+          "value": {
+            "amount": 11000000,
+            "currency": "GBP"
+          },
+          "items": [
+            {
+              "id": "0001",
+              "description": "string",
+              "classification": {
+                "scheme": "CPV",
+                "id": "45233130",
+                "description": "Construction work for highways",
+                "uri": "http://cpv.data.ac.uk/code-45233130"
+              },
+              "quantity": 8,
+              "unit": {
+                "name": "Miles",
+                "value": {
+                  "amount": 137000,
+                  "currency": "GBP"
+                }
+              }
+            }
+          ],
+          "dateSigned": "2018-01-10T14:23:12Z",
+          "implementation": {
+            "milestones": [
+              {
+                "id": "IMS/1-contract/1-implementation-milestones/1",
+                "title": "project started",
+                "type": "delivery",
+                "code": "commencement",
+                "dueDate": "2018-03-19T00:00:00+05:45",
+                "status": "met",
+                "dateMet": "2018-03-31T00:00:00+05:45",
+                "dateModified": "2018-03-31T00:00:00+05:45"
+              },
+              {
+                "id": "IMS/1-contract/1-implementation-milestones/2",
+                "title": "project completed",
+                "type": "delivery",
+                "code": "completion",
+                "dueDate": "2018-12-31T00:00:00+05:45",
+                "status": "notMet",
+                "dateModified": "2018-03-31T00:00:00+05:45"
```

Comment:
I'm confused by this example - how can we know for sure on 2018-03-31 that the completion due on 2018-12-31 will be not met? Or is this because it started late, it is now expected to finish late?

Also, it's not clear to me if examples 2 and 3 are intended to be from the same project, or are they 2 different possible scenarios following on from the first? Presumably since 3 finishes early, it's a different possible scenario?

Reply:
this is based on the Dhangadhi advice so maybe @duncandewhurst can clarify?

Reply:
> I'm confused by this example - how can we know for sure on 2018-03-31 that the completion due on 2018-12-31 will be not met? Or is this because it started late, it is now expected to finish late?

Based on the description of `milestone.dateModified` in the schema ('The date the milestone was last reviewed or modified and the status was altered or confirmed to still be correct.') my understanding is that `milestone.status` should be updated (or confirmed to be correct) each time the milestone is reviewed, so in this case the milestone was reviewed on 2018-03-31 and at that time it wasn't yet met (as expected, based on the `dueDate`) so the status is set to 'notMet', if in a future review the milestone is found to be met, then the status can be updated to 'met'

> Also, it's not clear to me if examples 2 and 3 are intended to be from the same project, or are they 2 different possible scenarios following on from the first? Presumably since 3 finishes early, it's a different possible scenario?

They are from the same contracting process, the project starts late but finishes early (very efficient of them!)

There is some introductory text at the top of the example in the Dhangadhi doc which would be good to adapt for inclusion in the worked example in the docs:

> This worked example shows how milestones for project commencement and project completion should be updated in OCDS.
>
> The example includes three updates to a single contracting process (ocds-XXXXXX-example):
> 
> The example is available in JSON and Google Sheets formats. Note that the example includes three updates to the same contracting process in a single JSON file/spreadsheet, however for Dhangadhi each update would replace the previous update, since only the latest version of the data about project is stored in the IMS system.
> 
> The following table explains what each update represents and shows the milestones in JSON format: